frontend/src/lib/utils.tsx

```diff
@@ -1325,4 +1328,39 @@ export function getEventNamesForAction(actionId: string | number, allActions: Ac
         .flatMap((a) => a.steps?.filter((step) => step.event).map((step) => String(step.event)) as string[])
 }
 
+export function isPropertyGroup(
+    properties: PropertyGroupFilter | AnyPropertyFilter[] | undefined
+): properties is PropertyGroupFilter {
+    if (properties) {
+        return (properties as PropertyGroupFilter).type !== undefined
+    }
+    return false
+}
+
+export function convertPropertiesToPropertyGroup(
+    properties: PropertyGroupFilter | AnyPropertyFilter[]
+): PropertyGroupFilter {
+    if (isPropertyGroup(properties)) {
+        return properties
+    }
+    return { type: FilterLogicalOperator.And, values: [{ type: FilterLogicalOperator.And, values: properties }] }
```

Comment:
We may want to revisit `PropertyGroupFilter` <> `PropertyGroupFilterValue` because this doesn't need to nest further here. Should ideally just return ``
```suggestion
    return { type: FilterLogicalOperator.And, values: properties }
```

Reply:
So for backwards compatible properties, can just pass in `properties` to `values` instead of the new format with `type`? It does make that part easier but introduces another "option" of formatting

Reply:
Eh I think ignore what I'm suggesting it's unnecessary